homeassistant/components/tradfri/cover.py

```diff
@@ -0,0 +1,149 @@
+"""Support for IKEA Tradfri covers."""
+import logging
+
+from pytradfri.error import PytradfriError
+
+from homeassistant.components.cover import (
+    CoverDevice,
+    ATTR_POSITION,
+    SUPPORT_OPEN,
+    SUPPORT_CLOSE,
+    SUPPORT_SET_POSITION,
+)
+from homeassistant.core import callback
+from . import DOMAIN as TRADFRI_DOMAIN, KEY_API, KEY_GATEWAY
+from .const import CONF_GATEWAY_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Load Tradfri covers based on a config entry."""
+    gateway_id = config_entry.data[CONF_GATEWAY_ID]
+    api = hass.data[KEY_API][config_entry.entry_id]
+    gateway = hass.data[KEY_GATEWAY][config_entry.entry_id]
+
+    devices_commands = await api(gateway.get_devices())
+    devices = await api(devices_commands)
+    covers = [dev for dev in devices if dev.has_blind_control]
+    if covers:
+        async_add_entities(TradfriCover(cover, api, gateway_id) for cover in covers)
+
+
+class TradfriCover(CoverDevice):
+    """The platform class required by Home Assistant."""
+
+    def __init__(self, cover, api, gateway_id):
+        """Initialize a cover."""
+        self._api = api
+        self._unique_id = f"cover-{gateway_id}-{cover.id}"
```

Comment:
The entity registry is aware of the integration domain and platform domain of the entity, so adding `cover` isn't necessary.

Reply:
I'll remove it but just a double check as I saw this in `light` and also noticed who made the change, so I figured it wasn't put there randomly?

![image](https://user-images.githubusercontent.com/21142447/65391696-79222c80-dd6c-11e9-9645-a38cd2400da8.png)

Reply:
The light platform has two kinds of entities, light and group. If there are more than one type of entity in a platform and the rest of the id is the same between the two kinds, we need to add a differentiator.

I don't know if this is a actually the case for the light platform, ie if `group.id` and `light.id` are the same.

Reply:
That makes sense, thanks for the explanation